ui/app.js

```diff
@@ -1052,6 +1053,42 @@ $(document).ready(() => {
     }
   });
 
+  $(".leaderboard-title").click(function () {
+    if (canSearchForOfficer == true) {
+      if ($(".leaderboard-search-input").css("display") == "none") {
+        $(".leaderboard-search-input").slideDown(250);
+        $(".leaderboard-search-input").css("display", "block");
+      } else {
+        $(".leaderboard-search-input").slideUp(250);
+        setTimeout(() => {
+          $(".leaderboard-search-input").css("display", "none");
+        }, 250);

```

Comment:
If I recall correct, the `slideDown` actually takes a function as a second parameter which triggers, when the animation is done. So you might be able to do something like:
```suggestion
        $(".leaderboard-search-input").slideUp(250, () => {
            $(".leaderboard-search-input").css("display", "none");
        });
```

Try giving it a go. The code would be way more clean that way.

Reply:
done <3